packages/template-retail-react-app/app/commerce-api/auth.js

```diff
@@ -139,23 +140,44 @@ class Auth {
         this._storage.set(oidStorageKey, oid)
     }
 
+    get isTokenValid() {
+        return (
+            isTokenValid(this.authToken) &&
+            !hasSFRAAuthStateChanged(this._storage, this._storageCopy)
+        )
+    }
+
     /**
      * Save refresh token in designated storage.
      *
      * @param {string} token The refresh token.
      * @param {USER_TYPE} type Type of the user.
      */
     _saveRefreshToken(token, type) {
+        /**
+         * For hybrid deployments, We store a copy of the refresh_token
+         * to update access_token whenever customer auth state changes on SFRA.
+         */
         if (type === Auth.USER_TYPE.REGISTERED) {
             this._storage.set(refreshTokenRegisteredStorageKey, token, {
                 expires: REFRESH_TOKEN_COOKIE_AGE
             })
             this._storage.delete(refreshTokenGuestStorageKey)
+
+            this._storageCopy.set(refreshTokenRegisteredStorageKey, token, {
+                expires: REFRESH_TOKEN_COOKIE_AGE
+            })
```

Comment:
expires is for cookies, it doesn't work for localstorage

Reply:
Dumb question: What does this look like if both the original and copy are stored in local store? They're stored separately still correct? Otherwise, since they have the same name, one would override the other.

Reply:
@vcua-mobify The original and copy would both point to LocalStorage only in case of standalone setups in which case the token values will be the same and `hasSFRAAuthStateChanged` does not check values it only compares the keys. which again will be the same in this case. So we should be good.

Reply:
I think we want to be actively comparing the values since we open ourselves up to an edge case if we just look at the keys.

On a PWA Kit only environment (the only scenario where both point the original and copy are in local store) I think it's fine if we compare the value with itself.

On a hybrid environment, the original will be in the cookie store while the copy is in local store so we should have different values then.